llvm/unittests/TargetParser/RISCVISAInfoTest.cpp

```diff
@@ -663,6 +663,32 @@ TEST(ParseArchString, RejectsConflictingExtensions) {
         ::testing::EndsWith(" is only supported for 'rv32'"));
   }
 
+  for (StringRef Input : {"rv32idc_xqciac0p3"}) {
+    EXPECT_EQ(toString(RISCVISAInfo::parseArchString(Input, true).takeError()),
+              "'xqciac' extension is incompatible with 'c' extension when 'd' "
+              "extension is enabled");
+  }
+
+  for (StringRef Input : {"rv32i_zcd_xqciac0p3"}) {
+    EXPECT_EQ(
+        toString(RISCVISAInfo::parseArchString(Input, true).takeError()),
+        "'xqciac' extension is incompatible with 'zcd' extension when 'd' "
+        "extension is enabled");
+  }
+
+  for (StringRef Input : {"rv32idc_xqcicm0p2"}) {
+    EXPECT_EQ(toString(RISCVISAInfo::parseArchString(Input, true).takeError()),
+              "'xqcicm' extension is incompatible with 'c' extension when 'd' "
+              "extension is enabled");
+  }
+
+  for (StringRef Input : {"rv32i_zcd_xqcicm0p2"}) {
+    EXPECT_EQ(
+        toString(RISCVISAInfo::parseArchString(Input, true).takeError()),
+        "'xqcicm' extension is incompatible with 'zcd' extension when 'd' "
+        "extension is enabled");
+  }
```

Comment:
Can we use `EXPECT_THAT(..., ::testing::EndsWith(...))` to make these loops have more than one expectation each? This is what we ended up using for a lot of the xqci tests fairly successfully.